x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plugin/TransportEsqlQueryAction.java

```diff
@@ -123,17 +133,16 @@ protected void doExecute(Task task, EsqlQueryRequest request, ActionListener<Esq
                 listener
             );
         } else {
-            // workaround for https://github.com/elastic/elasticsearch/issues/97916 - TODO remove this when we can
-            requestExecutor.execute(ActionRunnable.wrap(listener, l -> doExecuteForked(task, request, l)));
+            innerExecute(task, request, listener);
         }
     }
 
     @Override
     public void execute(EsqlQueryRequest request, EsqlQueryTask task, ActionListener<EsqlQueryResponse> listener) {
-        doExecuteForked(task, request, listener);
+        ActionListener.run(listener, l -> innerExecute(task, request, l));
```

Comment:
This is needed for the async case to ensure that the response/failure is routed through the passed in listener - as this listener coordinates between the waiting task and write-to-index task. It's largely benign for the sync case.